compair/api/login.py

```diff
@@ -156,17 +156,17 @@ def cas_auth():
                     third_party_type=ThirdPartyType.cas,
                     params=validation_response.attributes
                 )
+                thirdpartyuser.generate_or_link_user_account()
                 db.session.add(thirdpartyuser)
-
-            if not thirdpartyuser.user:
-                thirdpartyuser.generate_user_account()
+            elif not thirdpartyuser.user:
+                thirdpartyuser.generate_or_link_user_account()
 
             authenticate(thirdpartyuser.user, login_method=thirdpartyuser.third_party_type.value)
             thirdpartyuser.params = validation_response.attributes
 
             if sess.get('LTI') and sess.get('lti_create_user_link'):
                 lti_user = LTIUser.query.get_or_404(sess['lti_user'])
-                lti_user.compair_user = thirdpartyuser.user
+                lti_user.compair_user_id = thirdpartyuser.user_id
                 lti_user.upgrade_system_role()
                 lti_user.update_user_profile() # only one update_user_profile needed TODO: simplify this in #606
```

Comment:
Is the `TODO` still needed before closing #606?

Reply:
no, I think I covered what I wanted simplified in this PR and #760

Reply:
Actually I forgot to add an `upgrade_system_role` to for CAS/SAML login so I'll be adding that as well